charts/grafana/values.yaml

```diff
@@ -284,6 +285,27 @@ admin:
 # schedulerName:
 
 ## Extra environment variables that will be pass onto deployment pods
+##
+## to provide grafana with access to CloudWatch on AWS EKS:
+## 1. create an iam role of type "Web identity" with provider oidc.eks.* (note the provider for later)
+## 2. edit the "Trust relationships" of the role, add a line inside the StringEquals clause using the
+## same oidc eks provider as noted before (same as the existing line)
+## also, replace NAMESPACE and prometheus-operator-grafana with the service account namespace and name
+##
+##  "oidc.eks.us-east-1.amazonaws.com/id/XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX:sub": "system:serviceaccount:NAMESPACE:prometheus-operator-grafana",
+##
+## 3. attach a policy to the role, you can use a built in policy called CloudWatchReadOnlyAccess
+## 4. use the following env: (replace 123456789000 and iam-role-name-here with your aws account number and role name)
```

Comment:
This is handled by the pod admission hook automatically if it's setup only requiring the annotation to be present on the service account. I think this comment should mention these steps aren't commonly required.

maybe even link to the AWS eks docs

Reply:
I agree there are alternatives,
steps 1+2+3 can be replaced with an eksctl command
and steps 4+5 can be replaced with an admission hook if it's installed

I do feel these instructions should be present somewhere in their current form.

I'm okay with moving these notes to some other section/file if you feel values.yaml is not the place for this, possibly they'll be better in the service account section or a separate doc file.

I do believe these are important, mostly since a lot of EKS users don't have the permissions to create IAM roles themselves and admission hooks are not installed in vanilla EKS and sometimes aren't an option (for security/auditing reasons).